src/decrypter.js

```diff
@@ -58,108 +57,74 @@ ntoh = function(word) {
  * @constructor
  * @param key {Array} The key as an array of 4, 6 or 8 words.
  */
-AES = function (key) {
-  this._precompute();
-
-  var i, j, tmp,
-    encKey, decKey,
-    sbox = this._tables[0][4], decTable = this._tables[1],
-    keyLen = key.length, rcon = 1;
-
-  if (keyLen !== 4 && keyLen !== 6 && keyLen !== 8) {
-    throw new Error("Invalid aes key size");
-  }
-
-  encKey = key.slice(0);
-  decKey = [];
-  this._key = [encKey, decKey];
-
-  // schedule encryption keys
-  for (i = keyLen; i < 4 * keyLen + 28; i++) {
-    tmp = encKey[i-1];
-
-    // apply sbox
-    if (i%keyLen === 0 || (keyLen === 8 && i%keyLen === 4)) {
-      tmp = sbox[tmp>>>24]<<24 ^ sbox[tmp>>16&255]<<16 ^ sbox[tmp>>8&255]<<8 ^ sbox[tmp&255];
+class AES {
+  constructor(key) {
+   /**
+    * The expanded S-box and inverse S-box tables. These will be computed
+    * on the client so that we don't have to send them down the wire.
+    *
+    * There are two tables, _tables[0] is for encryption and
+    * _tables[1] is for decryption.
+    *
+    * The first 4 sub-tables are the expanded S-box with MixColumns. The
+    * last (_tables[01][4]) is the S-box itself.
+    *
+    * @private
+    */
+    this._tables = [[[], [], [], [], []], [[], [], [], [], []]];
+    this._precompute();
```

Comment:
I think changing it makes sense. Only precompute once, and only do it when/if we need to. Any other time AES is called we will use the `tables` that has already been precomputed.